### ISSUE TITLE
podman failing in Travis CI again, now with write /proc/self/attr/keycreate: invalid argument

### ISSUE THREAD

**Issue by adelton** [containers/libpod#2798]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Attempt to run `podman build` on Travis CI's Ubuntu Xenial fails with

```
container_linux.go:345: starting container process caused "process_linux.go:430: container init caused \"write /proc/self/attr/keycreate: invalid argument\"
```

**Steps to reproduce the issue:**

1. In `.travis.yml`, install podman using `sudo add-apt-repository -y ppa:projectatomic/ppa && sudo apt-get update -y && sudo apt-get install -y podman`
2. Run `sudo podman build` for some Dockerfile, for example using https://github.com/adelton/freeipa-container/tree/podman

**Describe the results you received:**

https://travis-ci.org/adelton/freeipa-container/jobs/512956513 failed with

```
STEP 4: RUN groupadd -g 288 kdcproxy ; useradd -u 288 -g 288 -c 'IPA KDC Proxy User' -d '/var/lib/kdcproxy' -s '/sbin/nologin' kdcproxy
error running container: error creating container for [/bin/sh -c groupadd -g 288 kdcproxy ; useradd -u 288 -g 288 -c 'IPA KDC Proxy User' -d '/var/lib/kdcproxy' -s '/sbin/nologin' kdcproxy]: container_linux.go:345: starting container process caused "process_linux.go:430: container init caused \"write /proc/self/attr/keycreate: invalid argument\"
```

**Describe the results you expected:**

The same build not failing, for example https://travis-ci.org/adelton/freeipa-container/jobs/508818334.

**Additional information you deem important (e.g. issue happens only occasionally):**

It is a regression.

**Output of `podman version`:**

No way to run `podman version` but the Travis CI log says

```
Unpacking cri-o-runc (1.0.0-rc6-1~ubuntu16.04.2~ppa74) ...
Selecting previously unselected package podman.
Preparing to unpack .../podman_1.2.0-1~ubuntu16.04.2~ppa14_amd64.deb ...
Unpacking podman (1.2.0-1~ubuntu16.04.2~ppa14) ...
Processing triggers for man-db (2.7.5-1) ...
Processing triggers for libc-bin (2.23-0ubuntu10) ...
Setting up conmon (0-1~dev~ubuntu16.04~ppa2) ...
Setting up containers-golang (0.1-1~dev~ubuntu16.04~ppa1) ...
Setting up containers-common (0.1.36-1~dev~ubuntu16.04.2~ppa9) ...
Setting up libgpgme11:amd64 (1.6.0-1) ...
Setting up containernetworking-plugins (0.7.3-1~ubuntu16.04.2~ppa2) ...
Setting up cri-o-runc (1.0.0-rc6-1~ubuntu16.04.2~ppa74) ...
Setting up podman (1.2.0-1~ubuntu16.04.2~ppa14) ...
```

**Output of `podman info --debug`:**

N/A

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Travis CI.

**Comment by rhatdan** [comment]
The updated runc package is causing this problem.

**Comment by giuseppe** [comment]
is this still happening?  It is fixed in upstream runc

**Comment by adelton** [comment]
Yes, the situation has improved: https://travis-ci.org/adelton/freeipa-container/jobs/512956506.

**Comment by giuseppe** [comment]
@adelton could we close this issue?

**Comment by adelton** [comment]
Could we use this as an opportunity to add some automated testing of podman setup/behaviour Travis CI, so that any potential regressions are caught and reported as soon as possible?

**Comment by giuseppe** [comment]
we stopped using travis for podman testing as the kernel there is too old and stuff like rootless user namespaces require a newer one.  Do you have anything in particular in mind on what we could test more?  We could probably add the test to the cirrus CI we are using right now

**Comment by adelton** [comment]
I want to help podman by testing freeipa-container not just with docker https://github.com/freeipa/freeipa-container/blob/master/.travis.yml but with podman as well: https://github.com/adelton/freeipa-container/blob/podman/.travis.yml. For my purposes Travis CI works fine.
But from time to time, the podman installation and behaviour on Xenial which I'm using on Travis CI gets broken. And I'd prefer for the podman team catching it early.

**Comment by rhatdan** [comment]
There is an updated runc that should fix this problem.

**Comment by marshallford** [comment]
I'm seeing this issue in the Fedora 29 official container image after a `dnf install -y podman`.

```
[root@podman /]# podman version
Version:            1.2.0
RemoteAPI Version:  1
Go Version:         go1.11.5
OS/Arch:            linux/amd64
```

**Comment by rhatdan** [comment]
Is SELinux enabled on your system?

**Comment by marshallford** [comment]
TLDR: No.

After trying `getenforce`, `cat /etc/sysconfig/selinux`, and `sestatus` I'm fairly confident that SELinux isn't even an option with Google's [Container-Optimized OS with containerd](https://cloud.google.com/kubernetes-engine/docs/concepts/using-containerd), which is the node image for the GKE cluster that I have been testing with. Based on my naive findings via the [K8s docs](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#assign-selinux-labels-to-a-container) the host OS must have SELinux enabled in order to set options within the container.

**Comment by rhatdan** [comment]
There is an updated version of runc that was just committed to upstream.   
@lsm5 Do we ship runc?  If yes then this fix should be in tomorrows repo.

Otherwise you could downgrade to the previous version of runc to remove the breakage,

**Comment by marshallford** [comment]
I've tried runc version 1.0.0-rc7 as well and get the same error as above. Are you referring to an upcoming 1.0.0-rc8 release?

**Comment by rhatdan** [comment]
Yes, I got the fix merged this morning, and they were waiting for the fix for rc8/

**Comment by marshallford** [comment]
I've confirmed that 1.0.0-rc8 fixes the issue I listed above. I will have to look into Fedora's packaging process and/or symlink runc to the the downloaded binary. @rhatdan Thank you for all of your help!

**Comment by rhatdan** [comment]
runc should be available in updates-testing and be released by early next week.

**Comment by gbraad** [comment]
No updates available on F29 yet

**Comment by marshallford** [comment]
@gbraad Same for F30. Just checked.

**Comment by rhatdan** [comment]
According to https://bodhi.fedoraproject.org/updates/FEDORA-2019-bc70b381ad it should be in stable now.  Well it reached that state yesterday.

**Comment by gbraad** [comment]
@rhatdan how about F29? I am not willing to update the whole system at this moment as I will present soon.

**Comment by rhatdan** [comment]
It should be there as well.

If it has not reached a repo yet, then just install it from updates-testing.

dnf -y update runc --enablerepo=updates-testing.
